tests/integration/test_core.py

```diff
@@ -255,3 +265,103 @@ async def test_wordpress_plugin_installation_error(
         assert (
                 unit.workload_status == ops.model.ActiveStatus.name
         ), "status should back to active after invalid plugin removed from config"
+
+
+@pytest.mark.asyncio
+async def test_openstack_object_storage_plugin(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name,
+        default_admin_password,
+        unit_ip_list,
+        openstack_environment
+):
+    """
+    arrange: after WordPress charm has been deployed, db relation established and openstack swift
+        server ready
+    act: update charm configuration for openstack object storage plugin
+    assert: openstack object storage plugin should be installed after the config update and
+        WordPress openstack swift object storage integration should be set up properly.
+    """
+
+    swift_conn = swiftclient.Connection(
```

Comment:
Is this worth being in a fixture?

Reply:
By design, when openstack swift integration is setup, there's no easy way to teardown the integration (all media files and posts using media files will be invalid). So this is always a one time thing.